graph-gateway/src/network/entities.rs

```diff
@@ -0,0 +1,196 @@
+    ///
+    /// This is, among all allocations associated with the indexer and deployment, the address
+    /// with the largest amount of allocated tokens.
+    pub largest_allocation: Address,
+    /// The indexer's indexing total allocated tokens.
+    ///
+    /// This is, the sum of all allocated tokens associated with the indexer and deployment.
+    pub total_allocated_tokens: u128,
+
+    /// The indexer
+    pub indexer: Arc<Indexer>,
+
+    /// The indexer's indexing status
+    pub status: Option<IndexingStatus>,
+    /// The indexer's indexing cost model
+    pub cost_model: Option<Ptr<CostModel>>,
+}
+
+/// The [`IndexingStatus`] struct represents the indexer's indexing status.
+#[derive(Debug, Clone)]
+pub struct IndexingStatus {
+    /// The latest block the indexer has indexed for the deployment.
+    pub latest_block: BlockNumber,
+    /// The minimum block the indexer has indexed for the deployment.
+    pub min_block: Option<BlockNumber>,
+}
+
+/// The [`Indexer`] struct represents an indexer in the network topology.
+///
+/// The indexer is a network  node that indexes a subgraph using one of its versions, i.e., one of
+/// the subgraph's deployments. The [`Indexing`] struct represents the indexer's indexing of a
+/// subgraph's deployment.
+#[derive(CustomDebug, Clone)]
+pub struct Indexer {
+    /// The indexer's ID.
+    pub id: Address,
+
+    /// The indexer's URL.
+    ///
+    /// It is guaranteed that the URL scheme is either HTTP or HTTPS and the URL has a host.
+    #[debug(with = Display::fmt)]
+    pub url: Url,
+
+    /// The indexer's "indexer service" version.
+    pub indexer_agent_version: Version,
+    /// The indexer's "graph node" version.
+    pub graph_node_version: Version,
+
+    /// Whether the indexer uses the legacy scalar.
+    pub legacy_scalar: bool,
+
+    /// The indexer's indexings set.
+    ///
+    /// It is a set of deployment IDs that the indexer is indexing.
+    pub indexings: HashSet<DeploymentId>,
+
+    /// The indexer's staked tokens.
+    pub staked_tokens: u128,
+}
+
+#[derive(Clone)]
+pub struct Subgraph {
+    /// Subgraph ID
+    pub id: SubgraphId,
+
+    /// The Subgraph ID on L2.
+    ///
+    /// It indicates that the subgraph has been transferred to L2, in that case it should not be
+    /// should not be served directly by this gateway.
+    pub l2_id: Option<SubgraphId>,
+    /// Whether the subgraph has been transferred to L2.
+    ///
+    /// If all associated deployments have been transferred to L2, it should not be served directly
+    /// by this gateway.
+    pub transferred_to_l2: bool,
+
+    /// The subgraph's chain name.
+    ///
+    /// This information is extracted from the highest version of the subgraph deployment's
+    /// manifest.
+    pub chain: String,
+    /// The subgraph's start block number.
+    ///
+    /// This information is extracted from the highest version of the subgraph deployment's
+    /// manifest.
+    pub start_block: BlockNumber,
```

Comment:
This field should not be associated with a subgraph, because it could be different at each deployment.

Reply:
Well... this value is used at this moment in the `client_query.rs` file, with the `manifest_min_block` name:

https://github.com/edgeandnode/gateway/blob/cef0c73d4d3be52a3220cc000ac628e6efc2cfc2/graph-gateway/src/client_query.rs#L317-L323

It is computed like this:

https://github.com/edgeandnode/gateway/blob/cef0c73d4d3be52a3220cc000ac628e6efc2cfc2/graph-gateway/src/client_query.rs#L246

This is basically what I stated in the field documentation above:

```rust
/// The subgraph's start block number.
///
/// This information is extracted from the highest version of the subgraph deployment's
/// manifest.
pub start_block: BlockNumber,
```

The only change here is that we are pre-computing it.

Reply:
ah, ok. I think I got confused between this field and the other one in the subgraph struct when scanning through.